src/connection/connection_v2.ts

```diff
@@ -22,16 +22,17 @@ export class ConnectionV2 extends BaseConnection {
     return this.options.account;
   }
 
-  private async getSystemEngineEndpoint(): Promise<string> {
+  private async getSystemEngineEndpointParameters(): Promise<
```

Comment:
nit: Perhaps `getSystemEngineEndpointAndParameters` would make the naming slightly better? Right now it reads as if it parses the url and returns parameters only.